lib/spack/spack/spec.py

```diff
@@ -1616,6 +1615,8 @@ def to_node_dict(self, hash=ht.dag_hash):
          """
         d = syaml.syaml_dict()
 
+        d['name'] = self.name
+
```

Comment:
Not directly here, but closest line to it.

In the parameters section, I think we should take this opportunity to stop writing compiler flag entries that are empty -- that will change all of the hashes in Spack, so we should do it at the same time as this PR that also changes them.

In `from_node_dict` we can just assume that any missing compiler flags are empty.

Reply:
When we reviewed this, we uncovered that these do, indeed, need to be written.